### PR TITLE
Developer documentation guide

### PR DESCRIPTION
**Pull request by SteveMutungi254**
Added a developer documentation guide covering:
- End-to-end process for contribution.
- Cmdlet best practices.
- Pipeline best practices.
- Parameter best practices

### CODE REVIEW

File: development-docs/design-guidelines/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Entra PowerShell Design Guidelines
+
+This article provides best practices and design guidelines for various components of Entra PowerShell
+
+## Table of contents
+
+- [Cmdlet Best Practices](./cmdlet-best-practices.md)
+  - [Cmdlet Naming Conventions](./cmdlet-best-practices.md#cmdlet-naming-conventions)
+- [Module Best Practices](./module-best-practices.md)
```
Comment (KenitoInc) [inline development-docs/design-guidelines/README.md:9]: Dead link

File: development-docs/design-guidelines/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  - [Module Metadata](./module-best-practices.md#module-metadata)
+  - [Module Dependencies](./module-best-practices.md#module-dependencies)
+- [Parameter Best Practices](./parameter-best-practices.md)
+  - [Parameter Guidelines](./parameter-best-practices.md#parameter-guidelines)
+    - [Parameter Naming Conventions](./parameter-best-practices.md#parameter-naming-conventions)
+    - [Parameter Types](./parameter-best-practices.md#parameter-types)
+  - [Parameter Set Guidelines](./parameter-best-practices.md#parameter-set-guidelines)
```
Comment (KenitoInc) [inline development-docs/design-guidelines/README.md:16]:
```suggestion
  - [Parameter Set Guidelines](./parameter-best-practices.md#parameter-sets-guidelines)
```

File: development-docs/design-guidelines/parameter-best-practices.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    [-OptionalParameterName <OptionalParameterValue>]
+    [-OptionalSwitchParameters]
+    [-OptionalParameterName] <RequiredParameterValue>
+```
+
+Using `New-Alias` cmdlet as an example:
+
+```powershell
+New-Alias 
+    [-Name] <string>                                                     -required 'positional' parameter
```
Comment (KenitoInc) [inline development-docs/design-guidelines/parameter-best-practices.md:130]:
```suggestion
    -Name <string>                                                     -required 'positional' parameter
```

File: development-docs/design-guidelines/cmdlet-best-practices.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+If your service or resource has a different name, or if you want to shorten the cmdlet name for easier recall, you can add an alias attribute to your cmdlet to enable this functionality.
+
+### Output Type
+
+`OutputType` attribute lets the user know what the type of the object returned by the cmdlet is (found in the **Outputs** section of a cmdlet's help content). The type specified here should always be a single element and not an enumeration of elements.
+
+#### Valid Output Types
+
+If the cmdlet returns an object, the type of the object returned must be defined; the output type for a cmdlet should _never_ be `object`, `PSObject`, `PSCustomObject` or the like. Returning these types of objects makes it difficult for the user to anticipate which properties will be found on the object returned from the cmdlet.
```
Comment (KenitoInc) [inline development-docs/design-guidelines/cmdlet-best-practices.md:51]: Do we need to state that we only return `MsGraph` types?

File: development-docs/README.md [hunk 5]
```diff
@@ -0,0 +1,190 @@
+# Entra PowerShell Developer Guide
+
+The Microsoft Entra PowerShell Developer Guide helps you develop and test Entra PowerShell cmdlets. It provides information on setting up your environment, implementing cmdlets, creating and running tests, and writing cmdlet documentation.
+
+## Table of Contents
+
+- [Table of Contents](#table-of-contents)
+- [Process-Overview](#process-overview)
```
[the remark this answers is not in the document]
Comment (SteveMutungi254) [reply development-docs/README.md:8]: Fixed.

[the remark this answers is not in the document]
Comment (SteveMutungi254) [reply development-docs/README.md]: Sorted.

File: development-docs/design-guidelines/README.md (hunk 1 above)
Comment (SteveMutungi254) [reply development-docs/design-guidelines/README.md:9]: Resolved.

File: development-docs/design-guidelines/cmdlet-best-practices.md (hunk 4 above)
Comment (SteveMutungi254) [reply development-docs/design-guidelines/cmdlet-best-practices.md]: Added some text at the end of the sentence.

**Review by KenitoInc — APPROVED** [review]

**Pull request merged**